grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/GrpcWebTrailersExtractor.java

```diff
@@ -141,40 +113,93 @@ protected HttpObject filter(HttpObject obj) {
                                 DecompressorRegistry.getDefaultInstance().lookupDecompressor(grpcEncoding);
                         if (decompressor == null) {
                             // Can't find decompressor.
-                            deframer.close();
+                            publisher.close();
                             return obj;
                         }
-                        deframer.decompressor(ForwardingDecompressor.forGrpc(decompressor));
+                        handler.decompressor(ForwardingDecompressor.forGrpc(decompressor));
                     }
-                    deframer.request(Integer.MAX_VALUE);
                     return obj;
                 }
 
-                if (obj instanceof HttpData && !deframer.isClosed()) {
+                if (obj instanceof HttpData && !publisher.isComplete()) {
                     final HttpData httpData = (HttpData) obj;
                     final HttpData wrapped = HttpData.wrap(
                             httpData.byteBuf(ByteBufAccessMode.RETAINED_DUPLICATE));
-                    deframer.deframe(wrapped, wrapped.isEndOfStream());
+                    final boolean ignored = publisher.tryWrite(wrapped);
                 }
                 return obj;
             }
 
             @Override
             protected void beforeComplete(Subscriber<? super HttpObject> subscriber) {
-                deframer.close();
+                publisher.close();
             }
 
             @Override
             protected Throwable beforeError(Subscriber<? super HttpObject> subscriber, Throwable cause) {
-                deframer.close();
+                publisher.close();
                 return cause;
             }
         };
         filteredHttpResponse.whenComplete().handle((unused, unused2) -> {
             // To make sure the deframer is closed even when the response is cancelled.
-            deframer.close();
+            publisher.close();
             return null;
         });
         return filteredHttpResponse;
     }
+
+    private static Subscriber<DeframedMessage> trailersSubscriber(ClientRequestContext ctx) {
+        return new Subscriber<DeframedMessage>() {
+            @Override
+            public void onSubscribe(Subscription s) {
+                s.request(Long.MAX_VALUE);
```

Comment:
Is this backpressure-safe?

Reply:
Looks like this doesn't have any backpressure issue. Could you add some comments about why we will not have a backpressure problem even if we do this here?

Reply:
The backpressure is controlled by HttpDeframe. I will leave a comment for that.